src/snovault/schema_utils.py

```diff
@@ -335,7 +335,12 @@ def validate_request(schema, request, data=None, current=None):
 
     validated, errors = validate(schema, data, current)
     for error in errors:
-        request.errors.add('body', list(error.path), error.message)
+        messages = list([x.message for x in error.context])
+        schema_paths = [list(x.absolute_schema_path) for x in error.context]
+        context = {}
+        for i in range(len(schema_paths)):
+          context["->".join([str(x) for x in schema_paths[i]])] = messages[i]
+        request.errors.add('body', list(error.path), error.message,json.dumps(context,indent=2))
```

Comment:
This doesn't look like it will work to me, since the `add` method of the request's `errors` class only takes 3 positional arguments (https://github.com/ENCODE-DCC/snovault/blob/master/src/snovault/validation.py#L33). This code also shouldn't JSON-encode the info because that will be taken care of by the response renderer.

Also, the paths we currently report are returned as lists so it would probably be good to be consistent rather than converting them to strings just for the context errors.

It would be better to add the errors while looping over error.context in a single pass rather than building two lists and then looping again. Some of the formatting also doesn't follow PEP8 (for example, 4-space indents and spaces after commas).

Here's how I would write it to fix all of that:

```python
for error in errors:
    error = error.copy()
    context = []
    for context_error in error.context:
        context.append({
          'name': list(context_error.absolute_schema_path),
          'description': context_error.message,
        })
    if context:
        request.errors.add(
            'body', list(error.path), error.message,
            context=context)
    else:
        request.errors.add(
          'body', list(error.path), error.message)
```

(This returns the context errors as a list of dicts rather than a dict mapping string paths to error messages, for better symmetry with how the top-level errors are formatted.)

Reply:
Thanks, David, for your review. I support you and your team in making the modifications that y'all see fit.